ingestion/src/metadata/utils/dbt_config.py

```diff
@@ -72,12 +73,21 @@ def _(config: DbtHttpConfig):
 @get_dbt_details.register
 def _(config: DbtS3Config):
     try:
+        prefix = ""
+        bucket_name = ""
+        if config.dbtPrefixConfig:
+            prefix = config.dbtPrefixConfig.dbtObjectPrefix
+            bucket_name = config.dbtPrefixConfig.dbtBucketName
         from metadata.utils.aws_client import AWSClient
 
         aws_client = AWSClient(config.dbtSecurityConfig).get_resource("s3")
-        buckets = aws_client.buckets.all()
+
+        if bucket_name == "":
+            buckets = aws_client.buckets.all()
+        else:
+            buckets = [aws_client.Bucket(bucket_name)]
         for bucket in buckets:
-            for bucket_object in bucket.objects.all():
+            for bucket_object in bucket.objects.filter(Prefix=prefix):
```

Comment:
I'd rewrite this a bit to avoid empty strings:

```python
def get_dbt_prefix_config(config) -> Tuple[Optional[str], Optional[str]]:
    """"
    Return (bucket, prefix) tuple
    """"
    if config.dbtPrefixConfig:
        return config.dbtPrefixConfig.dbtBucketName, config.dbtPrefixConfig.dbtObjectPrefix
    
    return None, None

def get_object_list(config) -> Iterable[str]:
    bucket_name, prefix = get_dbt_prefix_config(config)
    if not bucket_name:
        buckets = # fetch all buckets
    else:
        buckets = [# fetch one]
    
    for bucket in buckets:
        if prefix:
            obj_list = bucket.objects.filter(Prefix=prefix)
        else:
            obj_list = bucket.objects.all()
        for elem in obj_list:
            yield elem.key

# Then look for the file names in the returned generator
```

This is just from the top of my head, but wrapping the logic in a couple of functions and not abusing empty strings might be a bit cleaner to read and debug.

This is just some pseudo code, feel free to ignore it 

Thanks!

Reply:
I'll make use of the `get_dbt_prefix_config` method. 

Using a single method of `get_object_list` to get the objects for both s3 and gcs configs might not be possible since they both have different inbuilt methods to access the buckets and objects.